### DISCUSSION TITLE
Adjusting the log level settings in the gradle.properties file.

### DISCUSSION THREAD

**Discussion by ytakzk** [expo/expo#23053] [Archived: Questions] [answered]
My app encounters a heap size error (`java.lang.OutOfMemoryError: Java heap space`) while building an Android production build. To address this issue, I would like to modify the log level settings in the `gradle.properties` file. In other words, I want to include the following line in the `gradle.properties`:

```
org.gradle.logging.level=warn
```

Is it feasible to do this with Expo?

**Comment by Kudo** [comment, marked as answer]
hi there! you could try to use environment variables or command lines to archive it. reference: https://docs.gradle.org/current/userguide/build_environment.html

e.g.

```
GRADLE_OPTS='-Dorg.gradle.logging.level=debug' ./gradlew :app:assembleDebug
./gradlew -Dorg.gradle.logging.level=debug :app:assembleDebug
```

side node: for the `java.lang.OutOfMemoryError: Java heap space` error, you may try to increase the jvm memory limit:
https://docs.gradle.org/current/userguide/build_environment.html#sec:configuring_jvm_memory

**Comment by ytakzk** [comment]
@Kudo Thank you for your response.
Is there a way to configure them through a build plugin or build properties?

In the [Build Properties](https://docs.expo.dev/versions/latest/sdk/build-properties/#pluginconfigtypeandroid) documentation, I couldn't find any properties that allow altering the memory limit or log level.

Another potential solution could be creating a build plugin. However, as far as I'm aware, there might not be one available yet that can modify `gradle.properties` through a plugin.
https://docs.expo.dev/versions/latest/sdk/build-properties/#pluginconfigtypeandroid

**Reply by Kudo** [reply]
Hi @ytakzk, could you share your use case that cannot use environment variables or command lines rather than expo-build-properties?

**Reply by ytakzk** [reply]
@Kudo The reason for this is that our app is not using the bare workflow. Instead, we build it simply through `npx eas build --platform android --local` rather than using `./gradlew`. In this case, it would be helpful if these properties could also be specified in [PluginConfigTypeAndroid](https://docs.expo.dev/versions/latest/sdk/build-properties/#pluginconfigtypeandroid) or through the plugin files.

**Reply by Kudo** [reply]
@ytakzk for eas build, you could [specify environment variables](https://docs.expo.dev/eas-update/environment-variables/#setting-and-getting-environment-variables-when-building)
